src/dos/dos_locale_data.cpp

```diff
@@ -0,0 +1,2983 @@
+{
+	const std::string base_str = "COUNTRY_NAME_";
+	return base_str + country_code;
+}
+
+std::string KeyboardLayoutInfoEntry::GetMsgName() const
+{
+	assert(!layout_codes.empty());
+	const std::string base_str = "KEYBOARD_LAYOUT_NAME_";
+	auto layout_code = layout_codes[0];
+	upcase(layout_code);
+	return base_str + layout_code;
+}
+
+// ***************************************************************************
+// Code page (screen font) information
+// ***************************************************************************
+
+// Reference:
+// - https://gitlab.com/FreeDOS/base/cpidos/-/blob/master/DOC/CPIDOS/CODEPAGE.TXT
+
+// clang-format off
+const std::map<std::string, std::vector<uint16_t>> LocaleData::BundledCpiContent = {
+	{ "EGA.CPI",     { 437, 850, 852, 853, 857, 858 } },
+	{ "EGA2.CPI",    { 775, 859, 1116, 1117, 1118, 1119 } },
+	{ "EGA3.CPI",    { 771, 772, 808, 855, 866, 872 } },
+	{ "EGA4.CPI",    { 848, 849, 1125, 1131, 3012, 30010 } },
+	{ "EGA5.CPI",    { 113, 737, 851, 852, 858, 869 } },
+	{ "EGA6.CPI",    { 899, 30008, 58210, 59829, 60258, 60853 } },
+	{ "EGA7.CPI",    { 30011, 30013, 30014, 30017, 30018, 30019 } },
+	{ "EGA8.CPI",    { 770, 773, 774, 775, 777, 778 } },
+	{ "EGA9.CPI",    { 858, 860, 861, 863, 865, 867 } },
+	{ "EGA10.CPI",   { 667, 668, 790, 852, 991, 3845 } },
+	{ "EGA11.CPI",   { 858, 30000, 30001, 30004, 30007, 30009 } },
+	{ "EGA12.CPI",   { 852, 858, 30003, 30029, 30030, 58335 } },
+	{ "EGA13.CPI",   { 852, 895, 30002, 58152, 59234, 62306 } },
+	{ "EGA14.CPI",   { 30006, 30012, 30015, 30016, 30020, 30021 } },
+	{ "EGA15.CPI",   { 30023, 30024, 30025, 30026, 30027, 30028 } },
+	{ "EGA16.CPI",   { 858, 3021, 30005, 30022, 30031, 30032 } },
+	{ "EGA17.CPI",   { 862, 864, 30034, 30033, 30039, 30040 } },
+	{ "EGA18.CPI",   { 856, 3846, 3848 } },
+};
+// clang-format on
+
+// clang-format off
+const std::map<uint16_t, CodePageWarning> LocaleData::CodePageWarnings = {
+        { 60258, CodePageWarning::DottedI },
+};
+// clang-format on
+
+// ***************************************************************************
+// Keyboard layout information
+// ***************************************************************************
+
+// References:
+// - https://gitlab.com/FreeDOS/base/keyb_lay/-/tree/master/DOC/KEYB/LAYOUTS
+
+// NOTE: Default code pages for certain international layouts are set to 30023
+// (Southern Africa) or 30026 (Central Africa) - these were selected based on
+// the percentage of the language users (as checked in April 2024), see:
+// - https://en.wikipedia.org/wiki/EF_English_Proficiency_Index
+// - https://en.wikipedia.org/wiki/Geographical_distribution_of_French_speakers
+// - https://en.wikipedia.org/wiki/List_of_countries_and_territories_where_Spanish_is_an_official_language
+// - https://en.wikipedia.org/wiki/Portuguese_language (language usage map)
+
+// clang-format off
+const std::vector<KeyboardLayoutInfoEntry> LocaleData::KeyboardLayoutInfo = {
```

Comment:
This might be able to be `constexpr` as a std::array, and have the same compatible API calls.  Worth a try. 
I can't remember if `std::map` also became `constexpr` in C++20, but worth a try on those too.

Reply:
These data are used outside of this file - so they can't be `constexpr`'d.
And I don't want to move them to the header, as they are rather large - so I would prefer to ensure only one copy exists in the compiled code.

Reply:
Makes sense